sympy/calculus/util.py

```diff
@@ -1236,7 +1241,7 @@ def __lt__(self, other):
                 return True
             if self.min >= other.max:
                 return False
-        elif not(other.is_real or other is S.Infinity or
+        elif not(other.is_extended_real or other is S.Infinity or
```

Comment:
oo and -oo could be dropped. They are extended real.

Reply:
Thanks. I've removed them there and in the similar functions below.